ldap/servers/slapd/log.c

```diff
@@ -441,6 +441,23 @@ log_set_logging(const char *attrname, char *value, int logtype, char *errorbuf,
     return LDAP_SUCCESS;
 }
 
+void
+log_external_libs_debug_print(char *buffer)
+{
+    slapi_log_error(SLAPI_LOG_FATAL, "OpenLDAP Client", "%s", buffer);
```

Comment:
Shouldn't this be "external lib -> " rather than "openldap" since we could have many types of external lib?

Reply:
Log level should be TRACE or DEBUG maybe too?

Reply:
> Shouldn't this be "external lib -> " rather than "openldap" since we could have many types of external lib?

If we won't mention the library name then log messages like this one won't make sense:

```
[07/Dec/2020:10:43:58.042403434 -0500] - ERR - OpenLDAP Client - attempting to connect:
```

But I think you have a good point and I should give this function a more accurate name - `log_external_libs_debug_print` - probably something like `log_external_libs_debug_openldap_print`?

> Log level should be TRACE or DEBUG maybe too?

My idea of the usage is like this:
1. You have an issue;
2. You set the debug level you need (maybe only replication?);
3. You think that the problem happens somewhere in the middle of replica auth;
4. You enable nsslapd-external-libs-debug-enabled attribute;
5. You see only OpenLDAP and Replication level output (you don't need any other noise from TRACE or DEBUG)

The same way you can 'inject' the info in any other debug level.

So I think it makes sense to have it distinctive. So it's the reason I put it to the minimal level (FATAL).
As an option, we can put it to the separate loglevel, maybe?
What do you think?

Reply:
well I think fatal is just "too intense" - I think our default levels always display warn and error, so it could be better to set this to warn instead. 

But yes, now having read the other comments I agreed that the name should be "libldap/libber" instead of openldapclient in this, and that the function should be renamed to suit :)